### DISCUSSION TITLE
Security Considerations on the Open Internet

### DISCUSSION THREAD

**Discussion by pizzapanther** [photoprism/photoprism#2979] [Q&A] [answered]
Thinking of opening my PhotoPrism install to the open internet behind an Nginx Proxy or something similar. Any security considerations I should take into account? I see there is a Cloud Hosting option in the docs so I'm assuming PhotoPrism can handle an open internet environment?

**Comment by lastzero** [comment, marked as answer]
Generally yes, we run our own instances on the Internet and have no concerns. However, it depends on your expectations and the length of your password. There is a built-in request limit for failed logins (one per minute), but short passwords can still be "hacked" this way by brute force.

If you're using nginx, make sure websockets are enabled and buffering is disabled to allow video streaming.

**Reply by pizzapanther** [reply]
thanks for the tips!

**Reply by lastzero** [reply]
You're welcome! If you feel like you need anything else to make you feel comfortable with exposing your instance, let us know. Our Pro Edition has additional security features where you can adjust parameters and security headers are set automatically, but advanced users can also set them manually in the proxy configuration if they are using the Community Edition.

**Reply by pizzapanther** [reply]
been meaning to upgrade, this will force me to remember to do it :laughing:
